core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -227,9 +227,37 @@ class Ruleset {
         return stringList.joinToString()
     }
 
-
-    fun checkModLinks(): String {
+    /** Severity level of Mod RuleSet check */
+    enum class CheckModLinksStatus {OK, Warning, Error}
+    /** Result of a Mod RuleSet check */
+    // essentially a named Pair with a few shortcuts
+    class CheckModLinksResult(val status: CheckModLinksStatus, val message: String) {
+        // Empty constructor just makes the Complex Mod Check on the new game screen shorter 
+        constructor(): this(CheckModLinksStatus.OK, "")
+        // Constructor that joins lines
+        constructor(status: CheckModLinksStatus, lines: ArrayList<String>):
+                this (status,
+                    lines.joinToString("\n"))
+        // Constructor that auto-determines severity
+        constructor(warningCount: Int, lines: ArrayList<String>):
+                this (
+                    when {
+                        lines.isEmpty() -> CheckModLinksStatus.OK
+                        lines.size == warningCount -> CheckModLinksStatus.Warning
+                        else -> CheckModLinksStatus.Error
+                    },
+                    lines)
+        // Allows $this in format strings 
+        override fun toString() = message
+        // Readability shortcuts
+        val isError
```

Comment:
fun isError() = status == CheckModLinksStatus.Error

That way it's more obvious to the casual reader that this isn't a field saved in the class
In general I think that readonly param

Reply:
Chuckle. Yea, properties without backing fields aren't to be preferred 'just because we can'. And here the compiled result is identical anyway. Is this mayhaps related to me running the 'Darcula' theme with its garish colours? Sorry.
Their value is marginal anyway, I just wanted to have the call site as short as feasible...